core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -4497,7 +4497,12 @@ public CompletableFuture<TransactionConfidence> waitForConfirmation(Transaction
      * @return a future for an object that contains transaction confidence information
      */
     public CompletableFuture<TransactionConfidence> waitForConfirmations(Transaction tx, int requiredConfirmations) {
-        return tx.getConfidence().getDepthFuture(requiredConfirmations);
+
+        return getConfidence(tx).getDepthFuture(requiredConfirmations);
+    }
+
+    private TransactionConfidence getConfidence(Transaction tx) {
+        return tx.getConfidence(Context.get().getConfidenceTable());
```

Comment:
Why is the `Context.get()` added here?

Reply:
The other `getConfidence()` methods in `Transaction` eventually call the one that takes a `TxConfidenceTable`, so I wanted to call that one directly. And also prepare for not using `Transaction` to get the `TransactionConfidence`.

Reply:
I think this rationale could go into the commit comment.

Reply:
Done and force-pushed.